### PR TITLE
[1.11.x] Optaplanner quickstarts default branch should be `development`

### PR DESCRIPTION
**Pull request by radtriste**
and it should be used only as a fallback

### CODE REVIEW

File: .ci/jenkins/Jenkinsfile.prod.nightly [hunk 1]
```diff
@@ -180,11 +180,11 @@ def treatGitHashes(String gitInformationHashes) {
     return gitInformationHashes.replace('-', '.').replaceAll(/([\w\d\-\_\.]*\/)([\w\d\-\_\.]*)/,'$2.scm.hash').replace(';', '\n')
 }
 
-String getOptaPlannerBranch(String defaultBranch = '') {
+String getOptaPlannerBranch(String defaultBranch) {
     /* The OptaPlanner major version is shifted by 7 from the Kogito major version:
        Kogito 1.x.y -> OptaPlanner 8.x.y. */
     int majorVersionShift = 7
-    def branch = defaultBranch ?: env.CHANGE_BRANCH ?: BRANCH_NAME
+    def branch = env.CHANGE_BRANCH ?: BRANCH_NAME ?: defaultBranch
```
Comment (Ginxo) [inline .ci/jenkins/Jenkinsfile.prod.nightly]: the default branch will be never taken... so I guess we have to refactor the method to 
```
String getOptaPlannerBranch(String defaultBranch) {
    /* The OptaPlanner major version is shifted by 7 from the Kogito major version:
       Kogito 1.x.y -> OptaPlanner 8.x.y. */
    int majorVersionShift = 7
    def branch = env.CHANGE_BRANCH ?: BRANCH_NAME

    String [] branchSplit = branch.split("\\.")
    if (branchSplit.length == 3) {
        Integer optaplannerMajorVersion = Integer.parseInt(branchSplit[0]) + majorVersionShift
        return "${optaplannerMajorVersion}.${branchSplit[1]}.${branchSplit[2]}"
    } else {
       return defaultBranch
    }
}
```
and obviously always define `defaultBranch` argument by every method calling
wdyt?

**Review by Ginxo — CHANGES_REQUESTED** [review]
please review...

File: .ci/jenkins/Jenkinsfile.prod.nightly (hunk 1 above)
Comment (radtriste) [reply .ci/jenkins/Jenkinsfile.prod.nightly]: See my changes https://github.com/kiegroup/kogito-pipelines/pull/304/commits/861e29c9bed3bdf1c52cb229d63aeee217a7755c

**Review by Ginxo — APPROVED** [review]
LGTM, can you please cherry pick them for main branch (as you created the PR for it)?

**Comment by radtriste** [comment]
> LGTM, can you please cherry pick them for main branch (as you created the PR for it)?

done

**Comment by radtriste** [comment]
incorporated review from https://github.com/kiegroup/kogito-pipelines/pull/305

**Review by Ginxo — APPROVED** [review]
LGTM

**Pull request merged**